components/arm/client.go

```diff
@@ -56,7 +57,7 @@ func NewClientFromConn(
 }
 
 func (c *client) EndPosition(ctx context.Context, extra map[string]interface{}) (spatialmath.Pose, error) {
-	ext, err := protoutils.StructToStructPb(extra)
+	ext, err := data.GetExtraFromContext(ctx, extra)
 	if err != nil {
 		return nil, err
 	}
```

Comment:
Also discussed how to enforce that the error on line 69 doesn't get wrapped in some error that obscures the ErrNoCaptureToStore error. Couldn't really think of anything other than tests, which we could add but wouldn't enforce that this would be true for future component clients that we add. Maybe it's okay to not solve this for now.

Reply:
Is the concern here specifically about errors from other languages that get wrapped in some other error? In go, `errors.Is` and `errors.As` should check each error in the chain for a match [[0](https://go.dev/blog/go1.13-errors)]. For errors from other languages, can you remind me how we're defining them?

Reply:
No, I was actually just concerned about wrapping Go errors without using %w or errors.Wrap and us losing the error  type information.  I guess this is just a general issue in Go though, not really specific to Katherine's change. And at this point Go error wrapping is probably socialized enough that people know to do it. It was only added to the language a couple years ago and on the last project I worked on we frequently had issues where we weren't wrapping errors correctly and we lost that information. But I think I was just overthinking this :)

Reply:
Oh I see! Maybe just a PSA for wrapping Go errors correctly. I often err on the side of not making things out to be a user error, but it seems like if the errors weren't being wrapped correctly this _would_ actually be a user error, especially since the RDK code is publicly available. _If_ someone is having trouble with error comparison, they could reference the code and see that we use `errors.Is` and from the docs hopefully figure out how they should be wrapping their errors.

Reply:
Oh yes, actually the problem I was thinking of would be a programmer error (Viam engineer wraps error incorrectly, obscuring the user error message for a particular component), since Viam users should know from the documentation that in order to get their modular filter component to work, they have to return a specific grpc error to tell the data manager that the data was just filtered out. To answer your earlier question of how we're defining them, I think we're just going to have public error variables in each of the SDKs that construct the gRPC error. So my understanding is there will be an equivalent of `data. ErrNoCaptureToStore` in each of the other SDKs that uses the language-specific grpc error package to construct the correct grpc error.

Reply:
Ah okay, I see. Then I think the test or documentation (for Viam engineer) option should be sufficient. I am in favor of the documentation option, since we should be wrapping errors "correctly" anyway.

SGTM about the public error variables. Does this exist already? In the future, how should we communicate/coordinate changes to the error or new errors?

Reply:
Yep that's exactly right - each SDK will provide an error that constructs a grpc error equivalent to `data. ErrNoCaptureToStore`, python example [here](https://github.com/viamrobotics/viam-python-sdk/blob/f65ba277dc389b1597dc68d29457e203b8945e45/src/viam/errors.py#L98).

In terms of @dmhilly 's original worry about Viam engineers wrapping and returning this error without actually using `errors.Wrap` - the only other way I can think of handling this case is to do string parsing the way I had in a previous implementation (something like `strings.Contains(err.Error(), data.ErrNoCaptureToStore.Error())`), which we agreed was not great. My inclination is to accept this as a potential fail case for now - whenever we add a collector for a new component we can first check the client code to make sure there's no special wrapping of this error.

Reply:
Oops sorry didn't see your most recent comment @tahiyasalam - hopefully my comment answers a few of those qs!

Reply:
Yes, it does!